backend/app/routers/research.py

```
@@ -0,0 +1,409 @@
+        return JSONResponse(
+            status_code=502,
+            content={"detail": "Price source unavailable"},
+        )
+    except Exception as exc:  # noqa: BLE001 — generic 500 per CLAUDE.md
+        logger.error(
+            "Research price-history composition failed for position=%s "
+            "window=%s: %s",
+            position_id,
+            window,
+            exc,
+        )
```

Ruff (0.15.13)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)